pkg/controller/extension/add.go

```diff
@@ -63,12 +67,43 @@ func AddToManager(ctx context.Context, mgr manager.Manager) error {
 	return AddToManagerWithOptions(ctx, mgr, DefaultAddOptions)
 }
 
+// AddToManager2 adds a second controller with the default Options to the given Controller Manager.
+func AddToManager2(ctx context.Context, mgr manager.Manager) error {
+	return AddToManagerWithOptions2(ctx, mgr, DefaultAddOptions)
+}
+
 // AddToManagerWithOptions adds a controller with the given Options to the given manager.
 // The opts.Reconciler is being set with a newly instantiated actuator.
 func AddToManagerWithOptions(ctx context.Context, mgr manager.Manager, opts AddOptions) error {
 	predicates := extension.DefaultPredicates(ctx, mgr, DefaultAddOptions.IgnoreOperationAnnotation)
-	extensionClasses := []extensionsv1alpha1.ExtensionClass{extensionsv1alpha1.ExtensionClassShoot, extensionsv1alpha1.ExtensionClassSeed}
-	watchBuilder := extensionscontroller.NewWatchBuilder()
+
+	if opts.ExtensionClass == extensionsv1alpha1.ExtensionClassGarden {
+		return fmt.Errorf("controller %q for type %q is not supported for extension class %q", ControllerName, Type, opts.ExtensionClass)
+	}
+
+	extensionClasses := []extensionsv1alpha1.ExtensionClass{extensionsv1alpha1.ExtensionClassShoot}
+
+	return extension.Add(mgr, extension.AddArgs{
```

Comment:
Shouldn't the extension controller be separated into two distinct implementations?

Reply:
extension controllers for shoot and control plane have now separate implementations.